### PR TITLE
Fix: Switch from `torch.norm` to `torch.nn.functional.normalize`

### PR DESCRIPTION
**Pull request by KarahanS**
As stated in the issue #203, deprecated `torch.norm` should be replaced with `torch.nn.functional.normalize` to avoid any problems in the future. 
Made the change, run the unittests, everything works as expected.

### CODE REVIEW

**Review by monatis — APPROVED** [review]
Thanks for handling it.
